src/datasets/iterable_dataset.py

```diff
@@ -164,7 +182,11 @@ def shuffle_data_sources(self, seed: Optional[int]) -> "RandomlyCyclingMultiSour
 
 class MappedExamplesIterable(_BaseExamplesIterable):
     def __init__(
-        self, ex_iterable: _BaseExamplesIterable, function: Callable, batched: bool = False, batch_size: int = 1000
+        self,
+        ex_iterable: Union[_BaseExamplesIterable, Generator[Dict[str, Any], None, None]],
```

Comment:
Generators are iterators, and so have the same issue as in the previous comment.